javalib/src/main/scala/java/nio/file/attribute/FileTime.scala

```diff
@@ -3,15 +3,17 @@ package java.nio.file.attribute
 import java.util.concurrent.TimeUnit
 import java.time.Instant
 
-final class FileTime private (private val epochMillis: Long,
-                              private val nanos: Int)
+final class FileTime private (private val value: Long,
+                              private val unit: TimeUnit)
     extends Comparable[FileTime] {
 
   def compareTo(other: FileTime) = {
-    val compareMillis = epochMillis.compareTo(other.epochMillis)
-    if (compareMillis == 0) {
-      nanos.compareTo(other.nanos)
-    } else compareMillis
+    if (this.unit == other.unit) {
+      this.value.compareTo(other.value)
+    } else {
+      val compareUnit = TimeUnit.NANOSECONDS
+      this.to(compareUnit).compareTo(other.to(compareUnit))
```

Comment:
I don't think this will work in all cases. If one is stored in DAYS, its value in ns wouldn't contain all the information, and so it might compare equal to a `FileTime` that is stored in NANOSECONDS with that same value.

I think the easiest way to get this right is to compare *both* at the level of DAYS and NANOSECONDS. That would be enough, because 1 Day can be represented as a Long value in ns. Since DAYS and NANOSECONDS are the two extreme units, we would cover the whole range. So it would be something like
```suggestion
      val daysComp = this.to(TimeUnit.DAYS).compareTo(other.to(TimeUnit.DAYS))
      if (daysComp != 0)
        daysComp
      else
        this.to(TimeUnit.NANOSECONDS).compareTo(other.to(TimeUnit.NANOSECONDS))
```